ui/src/app/pages/analysis/runtime-panel.tsx

```diff
@@ -71,186 +73,201 @@ export interface Props {
+                    )(validLeonardoMachineTypes) }
+                value={memory}
+                />
+    </div>
+  </Fragment>;
+};
+
+const DiskSizeSelection = ({onChange, selectedDiskSize, currentRuntime}) => {
+  const {dataprocConfig, gceConfig} = currentRuntime;
+  const masterDiskSize = !!dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.bootDiskSize;
 
-export const RuntimePanel = fp.flow(withCurrentWorkspace(), withStore(runtimeOpsStore, 'runtimeOps'))(
-  class extends React.Component<Props, State> {
-    private aborter = new AbortController();
+  return <div>
+    <label htmlFor='runtime-disk'
+          style={{marginRight: '.25rem'}}>Disk (GB)</label>
+      <InputNumber id='runtime-disk'
+                //  disabled={true}
+                showButtons
+                decrementButtonClassName='p-button-secondary'
+                incrementButtonClassName='p-button-secondary'
+                value={selectedDiskSize || masterDiskSize}
+                inputStyle={{padding: '.75rem .5rem', width: '2rem'}}
+                onChange={({value}) => onChange(value === masterDiskSize ? null : value)}
+                min={50 /* Runtime API has a minimum 50GB requirement. */}/>
+  </div>;
+};
+
+export const RuntimePanel = withCurrentWorkspace()(({workspace}) => {
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(false);
+  const [selectedDiskSize, setSelectedDiskSize] = useState(null);
+  const [selectedMachine, setselectedMachine] = useState(null);
+  const runtimeOps = useStore(runtimeOpsStore);
+  const [currentRuntime, setRequestedRuntime] = useCustomRuntime(workspace.namespace);
 
-    constructor(props: Props) {
-      super(props);
-      this.state = {
-        loading: true,
-        error: false,
-        runtime: null
-      };
-    }
+  const activeRuntimeOp: RuntimeOperation = runtimeOps.opsByWorkspaceNamespace[workspace.namespace];
+  const {status = RuntimeStatus.Unknown, toolDockerImage = null, dataprocConfig = {}, gceConfig = {}} = currentRuntime || {};
+  const masterMachineType = !!dataprocConfig ? dataprocConfig.masterMachineType : gceConfig.machineType;
+  const masterDiskSize = !!dataprocConfig ? dataprocConfig.masterDiskSize : gceConfig.bootDiskSize;
+  const selectedMachineType = selectedMachine && selectedMachine.name;
 
-    async componentDidMount() {
-      // TODO(RW-5420): Centralize a runtimeStore.
-      let runtime = null;
-      let error = false;
+  useOnMount(() => {
+    const aborter = new AbortController();
+    const {namespace} = workspace;
+    const loadRuntime = async() => {
+        // TODO(RW-5420): Centralize a runtimeStore.
       try {
-        const promise = runtimeApi().getRuntime(this.props.workspace.namespace, {signal: this.aborter.signal});
-        updateRuntimeOpsStoreForWorkspaceNamespace(this.props.workspace.namespace, {
+        const promise = runtimeApi().getRuntime(namespace, {signal: aborter.signal});
```

Comment:
I wasn't expecting to see a get call here - shouldn't this just be using the store now?